js/main.js

```diff
@@ -1,55 +1,102 @@
 /* global data */
 /* exported data */
-
 var $form = document.querySelector('form');
 var $elements = $form.elements;
 var $photoUrl = $elements.photoUrl;
 var $img = document.getElementsByTagName('img')[0];
 var placeholder = $img.getAttribute('src');
 var $entryTitle = $elements.title;
 var $entryNotes = $elements.notes;
+var $ul = document.querySelector('ul');
+
+if (data.entries.length === 0) {
+  document.querySelector('.no-entries').className = 'no-entries';
+
+} else {
+  document.querySelector('.no-entries').className = 'no-entries hidden';
+}
 
-$photoUrl.addEventListener('input', handlePhoto);
 function handlePhoto(event) {
   if ($photoUrl.value === '') {
     $img.setAttribute('src', placeholder);
   } else {
     $img.setAttribute('src', $photoUrl.value);
   }
 }
+$photoUrl.addEventListener('input', handlePhoto);
 
-$form.addEventListener('submit', handleSubmit);
 function handleSubmit(event) {
   event.preventDefault();
   var $newEntry = {};
-  $newEntry[$entryTitle.name] = $entryTitle.value;
-  $newEntry[$photoUrl.name] = $photoUrl.value;
-  $newEntry[$entryNotes.name] = $entryNotes.value;
-  $newEntry.entryId = data.nextEntryId;
-  data.nextEntryId += 1;
-  data.entries.unshift($newEntry);
+  var len = data.entries.length;
+  if (data.entries.length === 0) {
+    $newEntry[$entryTitle.name] = $entryTitle.value;
+    $newEntry[$photoUrl.name] = $photoUrl.value;
+    $newEntry[$entryNotes.name] = $entryNotes.value;
+    $newEntry.entryId = data.nextEntryId;
+    data.nextEntryId += 1;
+    data.entries.unshift($newEntry);
+    $ul.prepend(entryTree($newEntry));
+  } else {
+    for (var i = 0; i < len; i++) {
+      if (data.editing !== null) {
+        if (String(data.entries[i].entryId) === String(data.editing.entryId)) {
+          var editCurrent = JSON.parse(JSON.stringify(data.editing));
+          editCurrent[$entryTitle.name] = $entryTitle.value;
+          editCurrent[$photoUrl.name] = $photoUrl.value;
+          editCurrent[$entryNotes.name] = $entryNotes.value;
+          $ul.replaceChild(entryTree(editCurrent), document.getElementById(data.editing.entryId));
+          data.editing = null;
+          break;
```

Comment:
You seem to be missing the portion where you find the original entry in `data.entries` and replace it with the new object after having been edited